File: Web/src/types/Tasks.ts
export interface Tasks {
  id: number;
  title: string;
  description: string;
  osId: OsId;
  addedDate: string;
}

export type OsId = 0| 1 | 3;

export const OperationalStatus: Record<OsId, string> = {
  0: "All",
  1: "Pending",
  3: "Completed",
};

export const OperationalStatusEnum: Record<string, OsId> = {
 All : 0,
 Pending: 1,
 Completed: 3,
};

File: DbLayer/Data/Models/Tasks.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DbLayer.Data.Models
{
	public class Tasks
	{

		[Key]
		public int Id                 { get; set; }

		public string Title            { get; set; }

		public string Description      { get; set; }

		public DateTime AddedDate               { get; set; }

		public DateTime? UpdatedDate            { get; set; }

		public int OsId                         { get; set; }
		
		[ForeignKey(nameof(OsId))]
		public OperationalStatus Status{ get; set; }
	}
}
